// pro-gallery-lib/dist/esm/settings/options/index.js
import itemClick from './itemClick';
import imageMargin from './imageMargin';
import hoveringBehaviour from './hoveringBehaviour';
import enableInfiniteScroll from './enableInfiniteScroll';
import cubeType from './cubeType';
import cubeImages from './cubeImages';
import cubeRatio from './cubeRatio';
import oneRow from './oneRow';
import titlePlacement from './titlePlacement';
import galleryMargin from './galleryMargin';
import isVertical from './isVertical';
import numberOfImagesPerRow from './numberOfImagesPerRow';
import numberOfImagesPerCol from './numberOfImagesPerCol';
import galleryTextAlign from './galleryTextAlign';
import videoPlay from './videoPlay';
import imageHoverAnimation from './imageHoverAnimation';
import overlayAnimation from './overlayAnimation';
import textImageSpace from './textImageSpace';
import textBoxBorderColor from './textBoxBorderColor';
import textBoxBorderRadius from './textBoxBorderRadius';
import textBoxBorderWidth from './textBoxBorderWidth';
import textBoxWidth from './textBoxWidth';
import textBoxWidthPercent from './textBoxWidthPercent';
import imageInfoType from './imageInfoType';
import groupSize from './groupSize';
import collageDensity from './collageDensity';
import gridStyle from './gridStyle';
import hasThumbnails from './hasThumbnails';
import groupTypes from './groupTypes';
import thumbnailSize from './thumbnailSize';
import galleryThumbnailsAlignment from './galleryThumbnailsAlignment';
import isRTL from './isRTL';
import scrollSnap from './scrollSnap';
import itemBorderWidth from './itemBorderWidth';
import itemBorderRadius from './itemBorderRadius';
import itemBorderColor from './itemBorderColor';
import rotatingCropRatios from './rotatingCropRatios';
import columnWidths from './columnWidths';
import autoSlideshowInterval from './autoSlideshowInterval';
import smartCrop from './smartCrop';
import minItemSize from './minItemSize';
import scrollAnimation from './scrollAnimation';
import slideAnimation from './slideAnimation';
import scatter from './scatter';
import rotatingScatter from './rotatingScatter';
import thumbnailSpacings from './thumbnailSpacings';
import slideshowLoop from './slideshowLoop';
import arrowsPadding from './arrowsPadding';
import arrowsSize from './arrowsSize';
import slideshowInfoSize from './slideshowInfoSize';
import textBoxHeight from './textBoxHeight';
import calculateTextBoxWidthMode from './calculateTextBoxWidthMode';
import chooseBestGroup from './chooseBestGroup';
import imageLoadingMode from './imageLoadingMode';
import cropOnlyFill from './cropOnlyFill';
import groupsPerStrip from './groupsPerStrip';
import fixedColumns from './fixedColumns';
import rotatingGroupTypes from './rotatingGroupTypes';
import arrowsPosition from './arrowsPosition';
import arrowsVerticalPosition from './arrowsVerticalPosition';
import itemShadowSize from './itemShadowSize';
import itemShadowBlur from './itemShadowBlur';
import itemShadowDirection from './itemShadowDirection';
import itemShadowOpacityAndColor from './itemShadowOpacityAndColor';
import itemEnableShadow from './itemEnableShadow';
import videoLoop from './videoLoop';
import showArrows from './showArrows';
import enableScroll from './enableScroll';
import imagePlacementAnimation from './imagePlacementAnimation';
import gallerySizeType from './gallerySizeType';
import gallerySize from './gallerySize';
import gallerySizePx from './gallerySizePx';
import gallerySizeRatio from './gallerySizeRatio';
import allowContextMenu from './allowContextMenu';
import galleryLayout from './galleryLayout';
import scrollDirection from './scrollDirection';
import scrollDuration from './scrollDuration';
import isAutoSlideshow from './isAutoSlideshow';
import loadMoreAmount from './loadMoreAmount';
import videoSound from './videoSound';
import videoSpeed from './videoSpeed';
import allowSlideshowCounter from './allowSlideshowCounter';
import playButtonForAutoSlideShow from './playButtonForAutoSlideShow';
import hidePlay from './hidePlay';
import overlayBackground from './overlayBackground';
import allowLeanGallery from './allowLeanGallery';
import placeGroupsLtr from './placeGroupsLtr';
import pauseAutoSlideshowOnHover from './pauseAutoSlideshowOnHover';
import showVideoControls from './showVideoControls';
import shouldIndexDirectShareLinkInSEO from './shouldIndexDirectShareLinkInSEO';
import slideTransition from './slideTransition';
import useMaxDimensions from './useMaxDimensions';
export default {
    itemClick: itemClick,
    imageMargin: imageMargin,
    hoveringBehaviour: hoveringBehaviour,
    enableInfiniteScroll: enableInfiniteScroll,
    cubeType: cubeType,
    cubeImages: cubeImages,
    cubeRatio: cubeRatio,
    oneRow: oneRow,
    titlePlacement: titlePlacement,
    galleryMargin: galleryMargin,
    isVertical: isVertical,
    numberOfImagesPerRow: numberOfImagesPerRow,
    numberOfImagesPerCol: numberOfImagesPerCol,
    galleryTextAlign: galleryTextAlign,
    videoPlay: videoPlay,
    imageHoverAnimation: imageHoverAnimation,
    overlayAnimation: overlayAnimation,
    textImageSpace: textImageSpace,
    textBoxBorderColor: textBoxBorderColor,
    textBoxBorderRadius: textBoxBorderRadius,
    textBoxBorderWidth: textBoxBorderWidth,
    textBoxWidth: textBoxWidth,
    textBoxWidthPercent: textBoxWidthPercent,
    imageInfoType: imageInfoType,
    groupSize: groupSize,
    collageDensity: collageDensity,
    gridStyle: gridStyle,
    hasThumbnails: hasThumbnails,
    groupTypes: groupTypes,
    thumbnailSize: thumbnailSize,
    galleryThumbnailsAlignment: galleryThumbnailsAlignment,
    isRTL: isRTL,
    scrollSnap: scrollSnap,
    itemBorderWidth: itemBorderWidth,
    itemBorderRadius: itemBorderRadius,
    itemBorderColor: itemBorderColor,
    rotatingCropRatios: rotatingCropRatios,
    columnWidths: columnWidths,
    autoSlideshowInterval: autoSlideshowInterval,
    smartCrop: smartCrop,
    minItemSize: minItemSize,
    scrollAnimation: scrollAnimation,
    slideAnimation: slideAnimation,
    scatter: scatter,
    rotatingScatter: rotatingScatter,
    thumbnailSpacings: thumbnailSpacings,
    slideshowLoop: slideshowLoop,
    arrowsPadding: arrowsPadding,
    arrowsSize: arrowsSize,
    slideshowInfoSize: slideshowInfoSize,
    textBoxHeight: textBoxHeight,
    calculateTextBoxWidthMode: calculateTextBoxWidthMode,
    chooseBestGroup: chooseBestGroup,
    imageLoadingMode: imageLoadingMode,
    cropOnlyFill: cropOnlyFill,
    groupsPerStrip: groupsPerStrip,
    fixedColumns: fixedColumns,
    rotatingGroupTypes: rotatingGroupTypes,
    arrowsPosition: arrowsPosition,
    arrowsVerticalPosition: arrowsVerticalPosition,
    itemShadowSize: itemShadowSize,
    itemShadowBlur: itemShadowBlur,
    itemShadowDirection: itemShadowDirection,
    itemShadowOpacityAndColor: itemShadowOpacityAndColor,
    itemEnableShadow: itemEnableShadow,
    videoLoop: videoLoop,
    showArrows: showArrows,
    enableScroll: enableScroll,
    imagePlacementAnimation: imagePlacementAnimation,
    gallerySizeType: gallerySizeType,
    gallerySize: gallerySize,
    gallerySizePx: gallerySizePx,
    gallerySizeRatio: gallerySizeRatio,
    allowContextMenu: allowContextMenu,
    galleryLayout: galleryLayout,
    scrollDirection: scrollDirection,
    scrollDuration: scrollDuration,
    isAutoSlideshow: isAutoSlideshow,
    loadMoreAmount: loadMoreAmount,
    videoSound: videoSound,
    videoSpeed: videoSpeed,
    allowSlideshowCounter: allowSlideshowCounter,
    playButtonForAutoSlideShow: playButtonForAutoSlideShow,
    hidePlay: hidePlay,
    overlayBackground: overlayBackground,
    allowLeanGallery: allowLeanGallery,
    placeGroupsLtr: placeGroupsLtr,
    showVideoControls: showVideoControls,
    pauseAutoSlideshowOnHover: pauseAutoSlideshowOnHover,
    shouldIndexDirectShareLinkInSEO: shouldIndexDirectShareLinkInSEO,
    slideTransition: slideTransition,
    useMaxDimensions: useMaxDimensions,
};
// TODO = add the options:
/*
itemOpacity
arrowsColor
imageLoadingColor
oneColorAnimationColor
*/
