// pro-gallery-lib/dist/cjs/core/presets/bricksGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var placements_1 = tslib_1.__importDefault(require("../../common/constants/placements"));
var scrollDirection_1 = tslib_1.__importDefault(require("../../common/constants/scrollDirection"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    galleryLayout: layout_1.default.BRICKS,
    cubeType: 'fill',
    cubeImages: true,
    cubeRatio: 1,
    titlePlacement: placements_1.default.SHOW_ON_HOVER,
    scrollDirection: scrollDirection_1.default.VERTICAL,
    galleryMargin: 0,
    isVertical: true,
    groupSize: 3,
    collageDensity: 0.8,
    groupTypes: '1,2h,2v,3t,3b,3l,3r,3v,3h',
    slideshowLoop: false,
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    gallerySize: 400,
    minItemSize: 50,
    chooseBestGroup: true,
    rotatingGroupTypes: '2h',
    smartCrop: false,
    scatter: 0,
    rotatingScatter: '',
    fixedColumns: 1,
    groupsPerStrip: 0,
    oneRow: false,
    placeGroupsLtr: false,
    rotatingCropRatios: '0.707,1.414,1.414,0.707',
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles) });
};
