// pro-gallery-lib/dist/cjs/core/presets/collageGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var placements_1 = tslib_1.__importDefault(require("../../common/constants/placements"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    galleryLayout: layout_1.default.COLLAGE,
    cubeImages: false,
    titlePlacement: placements_1.default.SHOW_ON_HOVER,
    groupSize: 3,
    hasThumbnails: false,
    groupTypes: '1,2h,2v,3t,3b,3l,3r',
    slideshowLoop: false,
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    targetItemSize: 0,
    fixedColumns: 0,
    enableScroll: true,
    isGrid: false,
    isSlider: false,
    isMasonry: false,
    isColumns: false,
    isSlideshow: false,
    cropOnlyFill: false,
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles, Math.round(styles.gallerySize * 5 + 500)) });
};
