// pro-gallery-lib/dist/cjs/core/presets/gridGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    galleryLayout: layout_1.default.GRID,
    cubeImages: true,
    isVertical: true,
    groupSize: 1,
    hasThumbnails: false,
    groupTypes: '1',
    slideshowLoop: false,
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    smartCrop: false,
    galleryType: 'Columns',
    fixedColumns: 0,
    targetItemSize: 0,
    enableScroll: true,
    cropOnlyFill: false,
    isSlider: false,
    isColumns: false,
    isGrid: true,
    isMasonry: false,
    isSlideshow: false,
    minItemSize: 50,
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles, Math.round(styles.gallerySize * 8.5 + 150)) });
};
