// pro-gallery-lib/dist/cjs/core/presets/panoramaGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var scrollDirection_1 = tslib_1.__importDefault(require("../../common/constants/scrollDirection"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    galleryLayout: layout_1.default.PANORAMA,
    cubeImages: false,
    scrollDirection: scrollDirection_1.default.VERTICAL,
    isVertical: true,
    groupSize: 1,
    groupTypes: '1',
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    galleryType: 'Columns',
    oneRow: false,
    fixedColumns: 1,
    enableScroll: true,
    isGrid: false,
    isColumns: false,
    isMasonry: false,
    isSlider: false,
    isSlideshow: false,
    cropOnlyFill: false,
    slideshowLoop: false,
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles) });
};
