// pro-gallery-lib/dist/cjs/core/presets/sliderGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var scrollDirection_1 = tslib_1.__importDefault(require("../../common/constants/scrollDirection"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    // tested params
    galleryLayout: layout_1.default.SLIDER,
    enableInfiniteScroll: true,
    cubeImages: true,
    oneRow: true,
    scrollDirection: scrollDirection_1.default.HORIZONTAL,
    isVertical: false,
    groupSize: 1,
    groupTypes: '1',
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    smartCrop: false,
    galleryType: 'Strips',
    hasThumbnails: false,
    enableScroll: true,
    scrollSnap: true,
    isGrid: false,
    isSlider: true,
    isColumns: false,
    isMasonry: false,
    isSlideshow: false,
    cropOnlyFill: true,
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles) });
};
