// pro-gallery-lib/dist/cjs/core/presets/thumbnailGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStyles = exports.fixedStyles = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
var placements_1 = tslib_1.__importDefault(require("../../common/constants/placements"));
var scrollDirection_1 = tslib_1.__importDefault(require("../../common/constants/scrollDirection"));
var layoutHelper_1 = require("../helpers/layoutHelper");
exports.fixedStyles = {
    galleryLayout: layout_1.default.THUMBNAIL,
    enableInfiniteScroll: true,
    cubeRatio: '100%/100%',
    cubeImages: true,
    oneRow: true,
    titlePlacement: placements_1.default.SHOW_ON_HOVER,
    scrollDirection: scrollDirection_1.default.HORIZONTAL,
    galleryMargin: 0,
    isVertical: false,
    groupSize: 1,
    groupTypes: '1',
    // this params were moved from the presets in layoutHelper and were not tested and checked yet.
    smartCrop: false,
    galleryType: 'Strips',
    hasThumbnails: true,
    enableScroll: true,
    scrollSnap: true,
    isGrid: false,
    isSlider: false,
    isMasonry: false,
    isColumns: false,
    isSlideshow: false,
    cropOnlyFill: false,
    scatter: 0,
    rotatingScatter: '',
    imageMargin: 0,
};
exports.createStyles = function (styles) {
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), exports.fixedStyles), { targetItemSize: layoutHelper_1.calcTargetItemSize(styles) });
};
