// pro-gallery-lib/dist/cjs/common/constants/placements.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isHorizontalPlacement = exports.isVerticalPlacement = exports.isLeftPlacement = exports.isRightPlacement = exports.isHoverPlacement = exports.isBelowPlacement = exports.isAbovePlacement = exports.hasHorizontalPlacement = exports.hasVerticalPlacement = exports.hasLeftPlacement = exports.hasRightPlacement = exports.hasHoverPlacement = exports.hasBelowPlacement = exports.hasAbovePlacement = void 0;
var PLACEMENTS = {
    SHOW_ON_HOVER: 'SHOW_ON_HOVER',
    SHOW_BELOW: 'SHOW_BELOW',
    SHOW_ABOVE: 'SHOW_ABOVE',
    SHOW_ON_THE_RIGHT: 'SHOW_ON_THE_RIGHT',
    SHOW_ON_THE_LEFT: 'SHOW_ON_THE_LEFT',
    ALTERNATE_HORIZONTAL: 'ALTERNATE_HORIZONTAL',
    ALTERNATE_VERTICAL: 'ALTERNATE_VERTICAL',
};
var hasHoverPlacement = function (placement) {
    return String(placement).indexOf(PLACEMENTS.SHOW_ON_HOVER) >= 0;
};
exports.hasHoverPlacement = hasHoverPlacement;
var hasAbovePlacement = function (placement, idx) {
    return String(placement).indexOf(PLACEMENTS.SHOW_ABOVE) >= 0 ||
        (idx % 2 === 0 &&
            String(placement).indexOf(PLACEMENTS.ALTERNATE_VERTICAL) >= 0);
};
exports.hasAbovePlacement = hasAbovePlacement;
var hasBelowPlacement = function (placement, idx) {
    return String(placement).indexOf(PLACEMENTS.SHOW_BELOW) >= 0 ||
        (idx % 2 === 1 &&
            String(placement).indexOf(PLACEMENTS.ALTERNATE_VERTICAL) >= 0);
};
exports.hasBelowPlacement = hasBelowPlacement;
var hasRightPlacement = function (placement, idx) {
    return String(placement).indexOf(PLACEMENTS.SHOW_ON_THE_RIGHT) >= 0 ||
        (idx % 2 === 0 &&
            String(placement).indexOf(PLACEMENTS.ALTERNATE_HORIZONTAL) >= 0);
};
exports.hasRightPlacement = hasRightPlacement;
var hasLeftPlacement = function (placement, idx) {
    return String(placement).indexOf(PLACEMENTS.SHOW_ON_THE_LEFT) >= 0 ||
        (idx % 2 === 1 &&
            String(placement).indexOf(PLACEMENTS.ALTERNATE_HORIZONTAL) >= 0);
};
exports.hasLeftPlacement = hasLeftPlacement;
var hasVerticalPlacement = function (placement) {
    return hasAbovePlacement(placement, 0) || hasBelowPlacement(placement, 1);
};
exports.hasVerticalPlacement = hasVerticalPlacement;
var hasHorizontalPlacement = function (placement) {
    return hasRightPlacement(placement, 0) || hasLeftPlacement(placement, 1);
};
exports.hasHorizontalPlacement = hasHorizontalPlacement;
var isVerticalPlacement = function (placement) {
    return hasVerticalPlacement(placement) &&
        !hasHorizontalPlacement(placement) &&
        !hasHoverPlacement(placement);
};
exports.isVerticalPlacement = isVerticalPlacement;
var isHorizontalPlacement = function (placement) {
    return hasHorizontalPlacement(placement) &&
        !hasVerticalPlacement(placement) &&
        !hasHoverPlacement(placement);
};
exports.isHorizontalPlacement = isHorizontalPlacement;
var isAbovePlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ABOVE;
};
exports.isAbovePlacement = isAbovePlacement;
var isBelowPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_BELOW;
};
exports.isBelowPlacement = isBelowPlacement;
var isHoverPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_HOVER;
};
exports.isHoverPlacement = isHoverPlacement;
var isRightPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_THE_RIGHT;
};
exports.isRightPlacement = isRightPlacement;
var isLeftPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_THE_LEFT;
};
exports.isLeftPlacement = isLeftPlacement;
exports.default = PLACEMENTS;
