// pro-gallery-lib/dist/cjs/common/coreStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = tslib_1.__importDefault(require("./constants"));
var coreStyles = {
    //duplicated from lib
    isRTL: false,
    isVertical: false,
    minItemSize: 120,
    groupSize: 3,
    chooseBestGroup: true,
    groupTypes: '1,2h,2v,3t,3b,3l,3r',
    cubeImages: false,
    cubeType: 'fill',
    smartCrop: false,
    fullscreen: true,
    videoLoop: true,
    videoSound: false,
    videoPlay: 'hover',
    videoSpeed: 1,
    numberOfImagesPerRow: 3,
    collageDensity: 0.8,
    galleryTextAlign: 'center',
    galleryMargin: 0,
    imageMargin: 10,
    fixedColumns: 0,
    cubeRatio: 1,
    showArrows: true,
    hasThumbnails: false,
    isSlideshow: false,
    galleryThumbnailsAlignment: 'bottom',
    gridStyle: 0,
    titlePlacement: constants_1.default.placements.SHOW_ON_HOVER,
    hoveringBehaviour: constants_1.default.infoBehaviourOnHover.APPEARS,
    isAutoSlideshow: false,
    slideshowLoop: false,
    playButtonForAutoSlideShow: false,
    pauseAutoSlideshowOnHover: true,
    allowSlideshowCounter: false,
    autoSlideshowInterval: 4,
    arrowsSize: 23,
    slideshowInfoSize: 200,
    imageLoadingMode: constants_1.default.loadingMode.BLUR,
    scrollAnimation: constants_1.default.scrollAnimations.NO_EFFECT,
    overlayAnimation: constants_1.default.overlayAnimations.NO_EFFECT,
    imageHoverAnimation: constants_1.default.imageHoverAnimations.NO_EFFECT,
    itemBorderWidth: 0,
    itemBorderRadius: 0,
    itemEnableShadow: false,
    itemShadowBlur: 20,
    loadMoreAmount: 'all',
    itemShadowDirection: 135,
    itemShadowSize: 10,
    imageInfoType: constants_1.default.infoType.NO_BACKGROUND,
    textBoxBorderRadius: 0,
    textBoxBorderWidth: 0,
    textBoxHeight: 200,
    textImageSpace: 10,
    scrollDirection: 0,
    slideAnimation: constants_1.default.slideAnimations.SCROLL,
    jsonStyleParams: '',
};
exports.default = coreStyles;
