// pro-gallery-lib/dist/cjs/common/defaultStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = tslib_1.__importDefault(require("./constants"));
var coreStyles_1 = tslib_1.__importDefault(require("./coreStyles"));
var defaultStyles = Object.assign({}, coreStyles_1.default, {
    gallerySize: 30,
    rotatingGroupTypes: '',
    cropOnlyFill: false,
    rotatingCropRatios: '',
    columnWidths: '',
    numberOfImagesPerCol: 1,
    groupsPerStrip: 0,
    imageMargin: 10,
    scatter: 0,
    rotatingScatter: '',
    placeGroupsLtr: false,
    oneRow: false,
    enableInfiniteScroll: true,
    thumbnailSpacings: 4,
    enableScroll: true,
    isGrid: false,
    isSlider: false,
    isColumns: false,
    isMasonry: false,
    scrollSnap: false,
    itemClick: constants_1.default.itemClick.NOTHING,
    slideAnimation: constants_1.default.slideAnimations.SCROLL,
    scrollDuration: 400,
    arrowsPosition: constants_1.default.arrowsPosition.ON_GALLERY,
    arrowsVerticalPosition: constants_1.default.arrowsVerticalPosition.ITEM_CENTER,
    arrowsPadding: 23,
    thumbnailSize: 120,
    magicLayoutSeed: 1,
    imagePlacementAnimation: constants_1.default.imagePlacementAnimations.NO_EFFECT,
    calculateTextBoxWidthMode: constants_1.default.textBoxWidthCalculationOptions.PERCENT,
    textBoxWidth: 200,
    textBoxWidthPercent: 50,
    loadMoreButtonText: '',
    videoSpeed: '1',
    showVideoControls: false,
    shouldIndexDirectShareLinkInSEO: true,
    slideTransition: constants_1.default.slideTransition.ease,
    useMaxDimensions: false,
});
exports.default = defaultStyles;
