// pro-gallery-lib/dist/cjs/settings/options/arrowsPosition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
exports.default = {
    title: 'Navigation Arrows Position',
    isRelevant: function (styleParams) { return styleParams.oneRow && styleParams.showArrows; },
    isRelevantDescription: 'Set a Horizontal gallery ("Scroll Direction" as "Horizontal") and set "Show Navigation Arrows" to "true".',
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.arrowsPosition.ON_GALLERY,
    options: utils_1.createOptions('arrowsPosition'),
    description: "Set the position of the navigation arrows in sliders. You can choose to position the arrows on the gallery (ON_GALLERY)\n  or outside the gallery (OUTSIDE_GALLERY- in these case the gallery will margins from both sides to make room for the arrows);\n  ",
};
