// pro-gallery-lib/dist/cjs/settings/options/calculateTextBoxWidthMode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
exports.default = {
    title: 'Text Box Width Units',
    description: "Set the text box width in pixels or as a percent from the column width.",
    isRelevantDescription: 'First set a Vertical gallery ("Scroll Direction" as "Vertical").\nThen set "Layout Orientation" to "Columns".\nThen set "Max Group Size" to "1".\nThen set "Texts Placement" to "Show on The Left" or "Show on The Right".',
    isRelevant: function (styleParams) {
        return styleParams.isVertical &&
            styleParams.groupSize === 1 &&
            !styleParams.oneRow &&
            (styleParams.titlePlacement ===
                constants_2.default.placements.SHOW_ON_THE_LEFT ||
                styleParams.titlePlacement ===
                    constants_2.default.placements.SHOW_ON_THE_RIGHT);
    },
    type: constants_1.INPUT_TYPES.OPTIONS,
    options: utils_1.createOptions('textBoxWidthCalculationOptions'),
    default: constants_2.default.textBoxWidthCalculationOptions.PERCENT,
};
