// pro-gallery-lib/dist/cjs/settings/options/imageInfoType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
exports.default = {
    title: 'Choose info layout',
    description: "Choose the layout you want for your texts, you can choose to separate the texts and the items\n  so you can style them separately.",
    isRelevantDescription: 'Set "Max Group Size" to "1", set "Layout Orientation" to "Columns" and set "Texts Placement" to anything but "Show On Hover".',
    isRelevant: function (styleParams) {
        return styleParams.groupSize === 1 &&
            styleParams.isVertical &&
            styleParams.titlePlacement !== constants_2.default.placements.SHOW_ON_HOVER;
    },
    options: utils_1.createOptions('infoType'),
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.infoType.NO_BACKGROUND,
};
