// pro-gallery-lib/dist/cjs/settings/options/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var itemClick_1 = tslib_1.__importDefault(require("./itemClick"));
var imageMargin_1 = tslib_1.__importDefault(require("./imageMargin"));
var hoveringBehaviour_1 = tslib_1.__importDefault(require("./hoveringBehaviour"));
var enableInfiniteScroll_1 = tslib_1.__importDefault(require("./enableInfiniteScroll"));
var cubeType_1 = tslib_1.__importDefault(require("./cubeType"));
var cubeImages_1 = tslib_1.__importDefault(require("./cubeImages"));
var cubeRatio_1 = tslib_1.__importDefault(require("./cubeRatio"));
var oneRow_1 = tslib_1.__importDefault(require("./oneRow"));
var titlePlacement_1 = tslib_1.__importDefault(require("./titlePlacement"));
var galleryMargin_1 = tslib_1.__importDefault(require("./galleryMargin"));
var isVertical_1 = tslib_1.__importDefault(require("./isVertical"));
var numberOfImagesPerRow_1 = tslib_1.__importDefault(require("./numberOfImagesPerRow"));
var numberOfImagesPerCol_1 = tslib_1.__importDefault(require("./numberOfImagesPerCol"));
var galleryTextAlign_1 = tslib_1.__importDefault(require("./galleryTextAlign"));
var videoPlay_1 = tslib_1.__importDefault(require("./videoPlay"));
var imageHoverAnimation_1 = tslib_1.__importDefault(require("./imageHoverAnimation"));
var overlayAnimation_1 = tslib_1.__importDefault(require("./overlayAnimation"));
var textImageSpace_1 = tslib_1.__importDefault(require("./textImageSpace"));
var textBoxBorderColor_1 = tslib_1.__importDefault(require("./textBoxBorderColor"));
var textBoxBorderRadius_1 = tslib_1.__importDefault(require("./textBoxBorderRadius"));
var textBoxBorderWidth_1 = tslib_1.__importDefault(require("./textBoxBorderWidth"));
var textBoxWidth_1 = tslib_1.__importDefault(require("./textBoxWidth"));
var textBoxWidthPercent_1 = tslib_1.__importDefault(require("./textBoxWidthPercent"));
var imageInfoType_1 = tslib_1.__importDefault(require("./imageInfoType"));
var groupSize_1 = tslib_1.__importDefault(require("./groupSize"));
var collageDensity_1 = tslib_1.__importDefault(require("./collageDensity"));
var gridStyle_1 = tslib_1.__importDefault(require("./gridStyle"));
var hasThumbnails_1 = tslib_1.__importDefault(require("./hasThumbnails"));
var groupTypes_1 = tslib_1.__importDefault(require("./groupTypes"));
var thumbnailSize_1 = tslib_1.__importDefault(require("./thumbnailSize"));
var galleryThumbnailsAlignment_1 = tslib_1.__importDefault(require("./galleryThumbnailsAlignment"));
var isRTL_1 = tslib_1.__importDefault(require("./isRTL"));
var scrollSnap_1 = tslib_1.__importDefault(require("./scrollSnap"));
var itemBorderWidth_1 = tslib_1.__importDefault(require("./itemBorderWidth"));
var itemBorderRadius_1 = tslib_1.__importDefault(require("./itemBorderRadius"));
var itemBorderColor_1 = tslib_1.__importDefault(require("./itemBorderColor"));
var rotatingCropRatios_1 = tslib_1.__importDefault(require("./rotatingCropRatios"));
var columnWidths_1 = tslib_1.__importDefault(require("./columnWidths"));
var autoSlideshowInterval_1 = tslib_1.__importDefault(require("./autoSlideshowInterval"));
var smartCrop_1 = tslib_1.__importDefault(require("./smartCrop"));
var minItemSize_1 = tslib_1.__importDefault(require("./minItemSize"));
var scrollAnimation_1 = tslib_1.__importDefault(require("./scrollAnimation"));
var slideAnimation_1 = tslib_1.__importDefault(require("./slideAnimation"));
var scatter_1 = tslib_1.__importDefault(require("./scatter"));
var rotatingScatter_1 = tslib_1.__importDefault(require("./rotatingScatter"));
var thumbnailSpacings_1 = tslib_1.__importDefault(require("./thumbnailSpacings"));
var slideshowLoop_1 = tslib_1.__importDefault(require("./slideshowLoop"));
var arrowsPadding_1 = tslib_1.__importDefault(require("./arrowsPadding"));
var arrowsSize_1 = tslib_1.__importDefault(require("./arrowsSize"));
var slideshowInfoSize_1 = tslib_1.__importDefault(require("./slideshowInfoSize"));
var textBoxHeight_1 = tslib_1.__importDefault(require("./textBoxHeight"));
var calculateTextBoxWidthMode_1 = tslib_1.__importDefault(require("./calculateTextBoxWidthMode"));
var chooseBestGroup_1 = tslib_1.__importDefault(require("./chooseBestGroup"));
var imageLoadingMode_1 = tslib_1.__importDefault(require("./imageLoadingMode"));
var cropOnlyFill_1 = tslib_1.__importDefault(require("./cropOnlyFill"));
var groupsPerStrip_1 = tslib_1.__importDefault(require("./groupsPerStrip"));
var fixedColumns_1 = tslib_1.__importDefault(require("./fixedColumns"));
var rotatingGroupTypes_1 = tslib_1.__importDefault(require("./rotatingGroupTypes"));
var arrowsPosition_1 = tslib_1.__importDefault(require("./arrowsPosition"));
var arrowsVerticalPosition_1 = tslib_1.__importDefault(require("./arrowsVerticalPosition"));
var itemShadowSize_1 = tslib_1.__importDefault(require("./itemShadowSize"));
var itemShadowBlur_1 = tslib_1.__importDefault(require("./itemShadowBlur"));
var itemShadowDirection_1 = tslib_1.__importDefault(require("./itemShadowDirection"));
var itemShadowOpacityAndColor_1 = tslib_1.__importDefault(require("./itemShadowOpacityAndColor"));
var itemEnableShadow_1 = tslib_1.__importDefault(require("./itemEnableShadow"));
var videoLoop_1 = tslib_1.__importDefault(require("./videoLoop"));
var showArrows_1 = tslib_1.__importDefault(require("./showArrows"));
var enableScroll_1 = tslib_1.__importDefault(require("./enableScroll"));
var imagePlacementAnimation_1 = tslib_1.__importDefault(require("./imagePlacementAnimation"));
var gallerySizeType_1 = tslib_1.__importDefault(require("./gallerySizeType"));
var gallerySize_1 = tslib_1.__importDefault(require("./gallerySize"));
var gallerySizePx_1 = tslib_1.__importDefault(require("./gallerySizePx"));
var gallerySizeRatio_1 = tslib_1.__importDefault(require("./gallerySizeRatio"));
var allowContextMenu_1 = tslib_1.__importDefault(require("./allowContextMenu"));
var galleryLayout_1 = tslib_1.__importDefault(require("./galleryLayout"));
var scrollDirection_1 = tslib_1.__importDefault(require("./scrollDirection"));
var scrollDuration_1 = tslib_1.__importDefault(require("./scrollDuration"));
var isAutoSlideshow_1 = tslib_1.__importDefault(require("./isAutoSlideshow"));
var loadMoreAmount_1 = tslib_1.__importDefault(require("./loadMoreAmount"));
var videoSound_1 = tslib_1.__importDefault(require("./videoSound"));
var videoSpeed_1 = tslib_1.__importDefault(require("./videoSpeed"));
var allowSlideshowCounter_1 = tslib_1.__importDefault(require("./allowSlideshowCounter"));
var playButtonForAutoSlideShow_1 = tslib_1.__importDefault(require("./playButtonForAutoSlideShow"));
var hidePlay_1 = tslib_1.__importDefault(require("./hidePlay"));
var overlayBackground_1 = tslib_1.__importDefault(require("./overlayBackground"));
var allowLeanGallery_1 = tslib_1.__importDefault(require("./allowLeanGallery"));
var placeGroupsLtr_1 = tslib_1.__importDefault(require("./placeGroupsLtr"));
var pauseAutoSlideshowOnHover_1 = tslib_1.__importDefault(require("./pauseAutoSlideshowOnHover"));
var showVideoControls_1 = tslib_1.__importDefault(require("./showVideoControls"));
var shouldIndexDirectShareLinkInSEO_1 = tslib_1.__importDefault(require("./shouldIndexDirectShareLinkInSEO"));
var slideTransition_1 = tslib_1.__importDefault(require("./slideTransition"));
var useMaxDimensions_1 = tslib_1.__importDefault(require("./useMaxDimensions"));
exports.default = {
    itemClick: itemClick_1.default,
    imageMargin: imageMargin_1.default,
    hoveringBehaviour: hoveringBehaviour_1.default,
    enableInfiniteScroll: enableInfiniteScroll_1.default,
    cubeType: cubeType_1.default,
    cubeImages: cubeImages_1.default,
    cubeRatio: cubeRatio_1.default,
    oneRow: oneRow_1.default,
    titlePlacement: titlePlacement_1.default,
    galleryMargin: galleryMargin_1.default,
    isVertical: isVertical_1.default,
    numberOfImagesPerRow: numberOfImagesPerRow_1.default,
    numberOfImagesPerCol: numberOfImagesPerCol_1.default,
    galleryTextAlign: galleryTextAlign_1.default,
    videoPlay: videoPlay_1.default,
    imageHoverAnimation: imageHoverAnimation_1.default,
    overlayAnimation: overlayAnimation_1.default,
    textImageSpace: textImageSpace_1.default,
    textBoxBorderColor: textBoxBorderColor_1.default,
    textBoxBorderRadius: textBoxBorderRadius_1.default,
    textBoxBorderWidth: textBoxBorderWidth_1.default,
    textBoxWidth: textBoxWidth_1.default,
    textBoxWidthPercent: textBoxWidthPercent_1.default,
    imageInfoType: imageInfoType_1.default,
    groupSize: groupSize_1.default,
    collageDensity: collageDensity_1.default,
    gridStyle: gridStyle_1.default,
    hasThumbnails: hasThumbnails_1.default,
    groupTypes: groupTypes_1.default,
    thumbnailSize: thumbnailSize_1.default,
    galleryThumbnailsAlignment: galleryThumbnailsAlignment_1.default,
    isRTL: isRTL_1.default,
    scrollSnap: scrollSnap_1.default,
    itemBorderWidth: itemBorderWidth_1.default,
    itemBorderRadius: itemBorderRadius_1.default,
    itemBorderColor: itemBorderColor_1.default,
    rotatingCropRatios: rotatingCropRatios_1.default,
    columnWidths: columnWidths_1.default,
    autoSlideshowInterval: autoSlideshowInterval_1.default,
    smartCrop: smartCrop_1.default,
    minItemSize: minItemSize_1.default,
    scrollAnimation: scrollAnimation_1.default,
    slideAnimation: slideAnimation_1.default,
    scatter: scatter_1.default,
    rotatingScatter: rotatingScatter_1.default,
    thumbnailSpacings: thumbnailSpacings_1.default,
    slideshowLoop: slideshowLoop_1.default,
    arrowsPadding: arrowsPadding_1.default,
    arrowsSize: arrowsSize_1.default,
    slideshowInfoSize: slideshowInfoSize_1.default,
    textBoxHeight: textBoxHeight_1.default,
    calculateTextBoxWidthMode: calculateTextBoxWidthMode_1.default,
    chooseBestGroup: chooseBestGroup_1.default,
    imageLoadingMode: imageLoadingMode_1.default,
    cropOnlyFill: cropOnlyFill_1.default,
    groupsPerStrip: groupsPerStrip_1.default,
    fixedColumns: fixedColumns_1.default,
    rotatingGroupTypes: rotatingGroupTypes_1.default,
    arrowsPosition: arrowsPosition_1.default,
    arrowsVerticalPosition: arrowsVerticalPosition_1.default,
    itemShadowSize: itemShadowSize_1.default,
    itemShadowBlur: itemShadowBlur_1.default,
    itemShadowDirection: itemShadowDirection_1.default,
    itemShadowOpacityAndColor: itemShadowOpacityAndColor_1.default,
    itemEnableShadow: itemEnableShadow_1.default,
    videoLoop: videoLoop_1.default,
    showArrows: showArrows_1.default,
    enableScroll: enableScroll_1.default,
    imagePlacementAnimation: imagePlacementAnimation_1.default,
    gallerySizeType: gallerySizeType_1.default,
    gallerySize: gallerySize_1.default,
    gallerySizePx: gallerySizePx_1.default,
    gallerySizeRatio: gallerySizeRatio_1.default,
    allowContextMenu: allowContextMenu_1.default,
    galleryLayout: galleryLayout_1.default,
    scrollDirection: scrollDirection_1.default,
    scrollDuration: scrollDuration_1.default,
    isAutoSlideshow: isAutoSlideshow_1.default,
    loadMoreAmount: loadMoreAmount_1.default,
    videoSound: videoSound_1.default,
    videoSpeed: videoSpeed_1.default,
    allowSlideshowCounter: allowSlideshowCounter_1.default,
    playButtonForAutoSlideShow: playButtonForAutoSlideShow_1.default,
    hidePlay: hidePlay_1.default,
    overlayBackground: overlayBackground_1.default,
    allowLeanGallery: allowLeanGallery_1.default,
    placeGroupsLtr: placeGroupsLtr_1.default,
    showVideoControls: showVideoControls_1.default,
    pauseAutoSlideshowOnHover: pauseAutoSlideshowOnHover_1.default,
    shouldIndexDirectShareLinkInSEO: shouldIndexDirectShareLinkInSEO_1.default,
    slideTransition: slideTransition_1.default,
    useMaxDimensions: useMaxDimensions_1.default,
};
// TODO = add the options:
/*
itemOpacity
arrowsColor
imageLoadingColor
oneColorAnimationColor
*/
