// pro-gallery-lib/dist/cjs/settings/options/scrollAnimation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
exports.default = {
    title: 'Scroll Animation',
    description: "Choose the type of animation to be used when items appear while scrolling verticaly through the gallery",
    isRelevantDescription: 'To enable "Scroll Animation" either set a Vertical gallery ("Scroll Direction" as "Vertical")\nor set a Horizontal gallery ("Scroll Direction" as "Horizontal") with "Slide Animation" set to "Scroll".',
    isRelevant: function (sp) {
        return !sp.oneRow ||
            (sp.oneRow && sp.slideAnimation === constants_2.default.slideAnimations.SCROLL);
    },
    options: utils_1.createOptions('scrollAnimations'),
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.scrollAnimations.NO_EFFECT,
};
