// pro-gallery-lib/dist/cjs/settings/options/textBoxWidth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
exports.default = {
    title: 'Text Box Width (pixels)',
    description: "Set the text box width when on the right side or on the left side.",
    isRelevantDescription: 'First set a Vertical gallery ("Scroll Direction" as "Vertical").\nThen set "Layout Orientation" to "Columns".\nThen set "Max Group Size" to "1".\nThen set "Texts Placement" to "Show on the Right" or "Show on the Left".\nThen set "Text Box Width Units" to "Manual".',
    isRelevant: function (styleParams) {
        return styleParams.isVertical &&
            styleParams.groupSize === 1 &&
            !styleParams.oneRow &&
            styleParams.calculateTextBoxWidthMode ===
                constants_2.default.textBoxWidthCalculationOptions.MANUAL &&
            constants_2.default.hasHorizontalPlacement(styleParams.titlePlacement);
    },
    type: constants_1.INPUT_TYPES.NUMBER,
    default: 200,
    min: 0,
    max: 1000,
};
