// pro-gallery-lib/dist/cjs/settings/options/titlePlacement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
exports.default = {
    title: 'Texts Placement',
    isRelevantDescription: 'Set a Vertical gallery ("Scroll Direction" as "Vertical"), set "Layout Orientation" to "Columns" and set "Max Group Size" to "1".',
    isRelevant: function (styleParams) {
        return (styleParams.isVertical || styleParams.oneRow) &&
            styleParams.groupSize === 1;
    },
    type: constants_1.INPUT_TYPES.MULTISELECT,
    default: constants_2.default.placements.SHOW_ON_HOVER,
    options: utils_1.createOptions('placements'),
    description: "Choose the the placement of the texts (title and description) relative to the items in the gallery. \n  Notes: \n   - this option also deals with the hover effects and may overide \"hoveringBehaviour\" when set to anything but \"SHOW_ON_HOVER\". \n   - you can select multiple values, but only one of each direction (ABOVE / BELOW, RIGHT / LEFT)\n  ",
};
